Assets/Scripts/saboteurManager.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class saboteurManager : MonoBehaviour
+{
+    List<ActionMessage> message_list = new List<ActionMessage>();
+    ActionMessage next_message = new ActionMessage();
+    ActionMessage persistMessage = new ActionMessage();
+    List<GuessMessage> guess_list = new List<GuessMessage>();
+    GuessMessage guess_message = new GuessMessage();
+
+    Dropdown step_dropdown;
+    Dropdown noun_dropdown;
+    Dropdown verb_dropdown;
+    Dropdown [] verb_Guess;
+    Dropdown [] noun_Guess;
+
+    private bool checkingForMessages = false;
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        step_dropdown = GameObject.Find("step_dropdown").GetComponent<Dropdown>();
+        verb_dropdown = GameObject.Find("verb_dropdown").GetComponent<Dropdown>();
+        noun_dropdown = GameObject.Find("noun_dropdown").GetComponent<Dropdown>();
+    
+        for(int i = 0; i < 4; i++)
+        {
+            verb_Guess[i] = GameObject.Find("VerbDropdownGuess " + "(" + i + ")").GetComponent<Dropdown>();
+            noun_Guess[i] = GameObject.Find("NounDropdownGuess " + "(" + i + ")").GetComponent<Dropdown>();
+
+        }
+
+        Debug.Log("Connecting to web socket");
```

Comment:
Wrap this with a conditional:

if (!NetworkMessaging.socketOpen())
        {
            NetworkMessaging.ConnectWebSocketToServerAsync("ws://localhost:8095/connectdemo");
        }